src/main/java/com/aws/greengrass/detector/config/Config.java

```diff
@@ -38,7 +51,11 @@ public Config(Topics topics) {
         configurationTopics.subscribe((whatHappened, node) -> {
             if (configurationTopics.isEmpty()) {
                 this.includeIPv4LoopbackAddrs = new AtomicBoolean(DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES);
```

Comment:
this defeats the purpose of atomics....

these need to be final fields which are just updated using `set`